### PR TITLE
Feature/globbing

### PR DESCRIPTION
**Pull request by scheul93**
Allows for globbing syntax in the src directory. This will allow inclusion or exclusion of certain files/directories.

Unit tests are not passing at the moment, Need to figure out how to mock globby

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,31 +31,53 @@ const run = config => {
...
-            });
+/**
+ * 
+ * @param {String} srcPath the full src path from the config
+ * This function finds the base path which is the path before any glob syntax is used
+ * This helps parse the file path in the copyFile method
+ * @returns {String} base path (the path before any glob syntx is used)
+ */
+function getSourceDir(srcPath) {
+    const srcPaths = srcPath.split('/');
```
Comment (heaper) [inline index.js]: The '/' path separator is only valid for Unix-based machines. Windows uses a different path separator.  Checkout path.sep and/or path.parse()

File: index.js [hunk 2]
```diff
@@ -29,31 +31,53 @@ const run = config => {
...
+    let retVal = srcPaths;
+    srcPaths.forEach((path, indx) => {
+        if (!glob.hasMagic(path)) {
+            const endIndx = indx + 1;
+            retVal = srcPaths.slice(0, endIndx).join('/');
```
Comment (heaper) [inline index.js]: See path separator comment above

File: index.js [hunk 3]
```diff
@@ -29,31 +31,53 @@ const run = config => {
...
+
+/**
+ *
+ * @param {Object} fileConfig config object from project
+ * this method gets all files in src (taking into account globbing syntax)
+ * it then calls the copyFile method to copy each file in directory
+ * @returns {Promise} a promise of all copyFile promises
+ */
+const copyDirectory = async (fileConfig) => {
+    fileConfig.basePath = getSourceDir(fileConfig.src);
```
Comment (heaper) [inline index.js]: I think you're mutating the configuration object that was passed in to this plugin from skeletor-core. I doubt this causes any problems, but in general it's a good idea to not mutate shared objects. Just wanted to mention that.

File: index.js [hunk 4]
```diff
@@ -29,31 +31,53 @@ const run = config => {
...
+    const filePaths = await glob(`${fileConfig.src}`);
```
Comment (heaper) [inline index.js]: This seems like an unnecessary template string. Can you just pass the variable directly?

Comment (scheul93) [reply index.js]: Noted. This was left over from a more complicated template string, so I'll clean this up.

**Pull request merged**